index.js

```diff
@@ -1,117 +1,142 @@
+  },
+  {
+    id: 3,
+    title: "Twilight",
+    author: "stephanie Meyer",
+    director: "this person",
+    genre: "this genre",
+  },
+];
 
-app.get('/movies/:director', (req, res) => {
-  res.json(topMovies.find((movies) => {
-  return movies.director ===req.params.director
-  }));
-  });
-  
-  app.get('/movies/:director/:genre', (req, res) => {
-    res.json(topMovies.find((movies) => {
-    return movies.genre ===req.params.genre
-    }));
-    });
-  
-  //adds data for a new movie to our list of movies. \\
-  
-  app.post('/movies', (req, res) => {
+app.get("/movies", (req, res) => {
+  res.json(topMovies);
+});
+
+//Get a single movie by title
+app.get("/movies/:title", (req, res) => {
+  res.json(
+    topMovies.find((movies) => {
+      return movies.title === req.params.title;
+    })
+  );
+});
+
+//Get list of movies by director
+app.get("/movies/director/:director", (req, res) => {
+  res.json(
+    topMovies.filter((movies) => {
+      return movies.director === req.params.director;
+    })
+  );
+});
+
+//Get list of movies by genre
+app.get("/movies/genre/:genre", (req, res) => {
+  res.json(
+    topMovies.filter((movies) => {
+      return movies.genre === req.params.genre;
+    })
+  );
+});
+
+//adds data for a new movie to our list of movies. \\
+app.post("/movies", (req, res) => {
   let newMovie = req.body;
-  
+
   if (!newMovie.name) {
-  const message = 'missing name in request body';
-  res.status(400).send(message);
+    const message = "missing name in request body";
+    res.status(400).send(message);
   } else {
-  newMovie.id = uuid.v4();
-  movies.push(newMovie);
-  res.status(201).send(newMovie);
-   }
-  });
-  
-  // deletes a movie from our list by ID \\
-  app.delete('/movies/:id', (req, res) => {
+    newMovie.id = uuid.v4();
+    movies.push(newMovie);
+    res.status(201).send(newMovie);
+  }
+});
+
+// deletes a movie from our list by ID \\
+app.delete("/movies/:id", (req, res) => {
   let movie = topMovies.find((movie) => {
-  return movie.id === req.params.id
+    return movie.id === req.params.id;
   });
-  
+
   if (movie) {
-  movie = movie.filter((obj) => {
-  return obj.id !== req.params.id 
-  });
-  res.status(201).send('movie ' + req.params.id + ' was deleted');
-  
-  }
-  });
-  
-  // make the "user" of a user \\
-  
-  app.put('/movies/:director/:genre/:imageURL/:user', (req, res) => {
-  let user = user.find((user) => {
-  return user.user === req.params.user
-  });
-  
-  if (user) {
-  user.user[req.params.user] = parseInt(req.params.user);
-  res.status(201).send('user ' + req.params.user + ' was added ' + req.params.grade + ' in ' + req.params.movies);
+    topMovies = topMovies.filter((obj) => {
+      return obj.id !== req.params.id;
+    });
+    res.status(201).send("movie " + req.params.id + " was deleted");
   } else {
-  res.status(404).send('user with the name ' + req.params.user + ' cannot be made.');
+    res.status(404).send("movie not found");
   }
-  });
```

Comment:
I wasn't sure if this endpoint was intended to be used to add a movie to the list of favorites of a user or to create a new user. So I recommend removing it and creating 2 separate endpoints as shown in this example.

- `app.post("/users", (req, res) => {` was added to allows the creation/registration of new users
- `app.put("/users/:id/movies/:movie_id"` was added to allow the addition of movies to the list of favorites of a user.